### PR TITLE
added get in front of URI Data object property methods and adapted tests accordingly

### PR DESCRIPTION
**Pull request by dkloepfer**

### CODE REVIEW

**Comment by dkloepfer** [comment]
@legionth Ping!

File: tests/Data/URITest.php [hunk 1]
```diff
@@ -503,17 +503,6 @@ public function test_with_host_invalid_1()
 		$uri->withHost('-foo-.de');
 	}
 
-	/**
-	 * @depends test_with_host
-	 * @expectedException \InvalidArgumentException
-	 */
-	public function test_with_host_invalid_2()
```
Comment (legionth) [inline tests/Data/URITest.php:510]: Why is this method removed? Is `null` now a valid value? What has this todo with the current changes?

**Review by legionth — CHANGES_REQUESTED** [review]
While the changes of the method names LGTM. I would suggest to open two separated PRs. Even if the changes are removing or refactoring tests(See `test_with_authority_invalid_2` and `test_with_host_invalid_2`).

**Review by klees — APPROVED** [review]
LGTM! Needs approval by JF, since this is a breaking change.

**Comment by matthiaskunkel** [comment]
Jour Fixe, 03 DEC 2018 : PR is highly appreciated. Please merge into trunk.

**Pull request merged**